tools/monorepo-utils/src/ci-jobs/lib/test-environment.ts

```diff
@@ -0,0 +1,201 @@
+/**
+ * External dependencies
+ */
+import https from 'node:http';
+
+/**
+ * Internal dependencies
+ */
+import { TestEnvConfigVars } from './config';
+
+/**
+ * The response for the WordPress.org stability check API.
+ */
+interface StableCheckResponse {
+	[ version: string ]: 'latest' | 'outdated' | 'insecure';
+}
+
+/**
+ * Gets all of the available WordPress versions and their associated stability.
+ *
+ * @return {Promise.<Object>} The response from the WordPress.org API.
+ */
+function getWordPressVersions(): Promise< StableCheckResponse > {
+	return new Promise< StableCheckResponse >( ( resolve, reject ) => {
+		// We're going to use the WordPress.org API to get information about available versions of WordPress.
+		const request = https.get(
+			'http://api.wordpress.org/core/stable-check/1.0/',
+			( response ) => {
+				// Listen for the response data.
+				let responseData = '';
+				response.on( 'data', ( chunk ) => {
+					responseData += chunk;
+				} );
+
+				// Once we have the entire response we can process it.
+				response.on( 'end', () =>
+					resolve( JSON.parse( responseData ) )
+				);
+			}
+		);
+
+		request.on( 'error', ( error ) => {
+			reject( error );
+		} );
+	} );
+}
+
+/**
+ * Uses the WordPress API to get the downlod URL to the latest version of an X.X version line. This
+ * also accepts "latest-X" to get an offset from the latest version of WordPress.
+ *
+ * @param {string} wpVersion The version of WordPress to look for.
+ * @return {Promise.<string>} The precise WP version download URL.
+ */
+async function getPreciseWPVersionURL( wpVersion: string ): Promise< string > {
+	const allVersions = await getWordPressVersions();
+
+	// If we're requesting a "latest" offset then we need to figure out what version line we're offsetting from.
+	const latestSubMatch = wpVersion.match( /^latest(?:-([0-9]+))?$/i );
```

Comment:
We probably don't want `latest-0`.

```suggestion
	const latestSubMatch = wpVersion.match( /^latest(?:-([1-9]+))?$/i );
```

Reply:
To go with my comment below, could limit this to 1-4 so the oldest is `latest-4`?

Reply:
While it's true that it isn't likely that anyone would use `latest-0` or `latest-{>3}`, I don't think there's a need to add extra handling here to check that. Arbitrarily high offsets are supported safely and although odd, `latest-0` doesn't hurt anything. I'd prefer to avoid adding more checks here without a clear benefit to limiting this.